File: plugins/ios/chat-module/Sources/StarterChat/Models/ChatSession.swift
import Foundation
import SwiftData

@Model
public final class ChatSession {
    public var id: UUID
    public var title: String
    public var createdAt: Date
    public var updatedAt: Date

    @Relationship(deleteRule: .cascade, inverse: \ChatMessage.sessionId)
    public var messages: [ChatMessage] = []

    public init(id: UUID = UUID(), title: String, createdAt: Date = Date(), updatedAt: Date = Date()) {
        self.id = id
        self.title = title
        self.createdAt = createdAt
        self.updatedAt = updatedAt
    }
}

File: plugins/flutter/chat/lib/src/models/chat_session.dart
/// Mirrors `ChatSession.swift` (the SwiftData `@Model`).
class ChatSession {
  const ChatSession({
    required this.id,
    required this.title,
    required this.createdAt,
    required this.updatedAt,
  });

  final String id;
  final String title;
  final DateTime createdAt;
  final DateTime updatedAt;

  ChatSession copyWith({String? title, DateTime? updatedAt}) => ChatSession(
    id: id,
    title: title ?? this.title,
    createdAt: createdAt,
    updatedAt: updatedAt ?? this.updatedAt,
  );
}
